config.go

```diff
@@ -42,11 +45,40 @@ func WithServerHandshaker(handshaker Handshaker) ServerOpt {
 
 // WithUnaryServerInterceptor sets the provided interceptor on the server
 func WithUnaryServerInterceptor(i UnaryServerInterceptor) ServerOpt {
+	return func(c *serverConfig) error {
+		if c.interceptor == nil {
```

Comment:
same comment to client interceptor :)

Reply:
I reverted to the original behavior (and updated the error message to be more accurate with these changes in place).